server/auth/sso/sso.go

```diff
@@ -2,27 +2,36 @@ package sso
 
 import (
 	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
 	"fmt"
 	"net/http"
 	"strings"
 	"time"
 
-	"github.com/argoproj/pkg/jwt/zjwt"
-	"github.com/argoproj/pkg/rand"
+	pkgrand "github.com/argoproj/pkg/rand"
 	"github.com/coreos/go-oidc"
 	log "github.com/sirupsen/logrus"
 	"golang.org/x/oauth2"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
 	apiv1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
-
-	"github.com/argoproj/argo/server/auth/jws"
 )
 
-const Prefix = "Bearer id_token:"
+const (
+	Prefix                              = "Bearer v2:"
+	issuer                              = "argo-server"                // the JWT issuer
+	expiry                              = 10 * time.Hour               // how long JWT are valid for
```

Comment:
Do we want this tunable without a new release being needed (e.g. environment variable)?

Reply:
yes - at some point - but not MVP